thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/completeness/checker/DataCompletenessTaskUtils.java

```diff
@@ -0,0 +1,218 @@
+
+    String timeFormat = timeSpec.getFormat();
+    if (timeFormat.equals(TimeSpec.SINCE_EPOCH_FORMAT)) {
+      TimeUnit unit = timeSpec.getDataGranularity().getUnit();
+      int timeDuration = timeSpec.getDataGranularity().getSize();
+
+      for (Entry<String, Long> entry : bucketNameToBucketValue.entrySet()) {
+        String bucketName = entry.getKey();
+        Long bucketValue = entry.getValue();
+        long timeValue = 0;
+
+        switch (unit) {
+          case MINUTES:
+            for (int i = 0; i < MINUTES_LEVEL_ROUNDING/timeDuration; i++) {
+              timeValue = TimeUnit.MINUTES.convert(bucketValue, TimeUnit.MILLISECONDS) / timeDuration;
+              bucketNameToTimeValues.put(bucketName, timeValue);
+              bucketValue = bucketValue + TimeUnit.MILLISECONDS.convert(timeDuration, TimeUnit.MINUTES);
+            }
+            break;
+          case DAYS:
+            timeValue = TimeUnit.DAYS.convert(bucketValue, TimeUnit.MILLISECONDS);
+            bucketNameToTimeValues.put(bucketName, timeValue);
+            break;
+          case HOURS:
+          default:
+            timeValue = TimeUnit.HOURS.convert(bucketValue, TimeUnit.MILLISECONDS);
+            bucketNameToTimeValues.put(bucketName, timeValue);
+            break;
+        }
+      }
+    } else {
+      for (Entry<String, Long> entry : bucketNameToBucketValue.entrySet()) {
+        String bucketName = entry.getKey();
+        bucketNameToTimeValues.put(bucketName, Long.valueOf(bucketName));
+      }
+    }
+    return bucketNameToTimeValues;
+  }
+
+  /**
+   * Get count * of buckets
+   * @param dataset
+   * @param bucketNameToBucketValue
+   * @return
+   */
+  public static Map<String, Long> getCountsForBucketsOfDataset(String dataset, TimeSpec timeSpec,
+      ListMultimap<String, Long> bucketNameToTimeValues) {
+    Map<String, Long> bucketNameToCountStar = new HashMap<>();
+
+    // generate request
+    StringBuilder sb = new StringBuilder();
+    String delimiter = "";
+    for (Long timeValue : bucketNameToTimeValues.values()) {
+      sb.append(delimiter);
+      delimiter = " OR ";
+      sb.append(String.format("%s='%s'", timeSpec.getColumnName(), timeValue));
+    }
+    long top = bucketNameToTimeValues.values().size();
+    String pql = String.format("select count(*) from %s where %s group by %s top %s",
+        dataset, sb.toString(), timeSpec.getColumnName(), top);
+    Map<Long, Long> timeValueToCount = new HashMap<>();
+    try {
+      ResultSetGroup resultSetGroup = CACHE_REGISTRY.getResultSetGroupCache().get(new PinotQuery(pql, dataset));
+      ResultSet resultSet = resultSetGroup.getResultSet(0);
+      for (int i = 0; i < resultSet.getRowCount(); i++) {
+        Long timeValue = Long.valueOf(resultSet.getGroupKeyString(i, 0));
+        Long count = resultSet.getLong(i, 0);
+        timeValueToCount.put(timeValue, count);
+      }
+
+    } catch (ExecutionException e) {
+      LOG.error("Exception in getting count *", e);
```

Comment:
Minor suggestion: Put more information to the log, e.g., "Exception in getting count * for dataset {}..."